client/main.js

```diff
@@ -0,0 +1,136 @@
+function httpGetAsync(theUrl, callback, message)
+{
+    var xmlHttp = new XMLHttpRequest();
+    xmlHttp.onreadystatechange = function() { 
+        if (xmlHttp.readyState == 4 && xmlHttp.status == 200)
+            callback(xmlHttp.responseText);
+    }
+    xmlHttp.open("POST", theUrl, true); // true for asynchronous 
+    xmlHttp.setRequestHeader("Content-Type", "application/json");
+    xmlHttp.send(JSON.stringify({ mfcc: message }));
+}
+
+function callback(res) {
+    console.log(res);
+}
+function doSetTimeout(a, i) {
+    setTimeout(function() {
+        var features = null;
+        features = a.get([
+            'mfcc'
+        ]);
+        console.log(features.mfcc)
+        httpGetAsync('http://127.0.0.1:80', callback, features.mfcc);
+    }, 500*i);
+}  
+
+function extractFeature(a) {
+    //console.log(a)
+    for (var i = 1; i <= 30; ++i) {
+        doSetTimeout(a, i);
+    }
+}
+
+class Audio {
+    constructor(bufferSize) {
+    that = this;
+    if (window.hasOwnProperty('webkitAudioContext') &&
+      !window.hasOwnProperty('AudioContext')) {
+        window.AudioContext = webkitAudioContext;
+      }
+
+    if (navigator.hasOwnProperty('webkitGetUserMedia') &&
+      !navigator.hasOwnProperty('getUserMedia')) {
+        navigator.getUserMedia = webkitGetUserMedia;
+        if (!AudioContext.prototype.hasOwnProperty('createScriptProcessor')) {
+          AudioContext.prototype.createScriptProcessor = AudioContext.prototype.createJavaScriptNode;
+        }
+      }
+
+    this.context = new AudioContext();
+    //console.log(this.context.sampleRate)
+
+    let elvis = document.getElementById('elvisSong');
+    let stream = this.context.createMediaElementSource(elvis);
+    stream.connect(this.context.destination);
+    this.meyda = Meyda.createMeydaAnalyzer({
+      audioContext: this.context,
+      source: stream,
+      bufferSize: bufferSize,
+      hopSize: 160,
+      callback: function(obj) {
+        alert("call back");
+      },
+    });
+    this.initializeMicrophoneSampling();
+  };
+
+  initializeMicrophoneSampling() {
+    var errorCallback = function (err) {
+      // We should fallback to an audio file here, but that's difficult on mobile
```

Comment:
falling back seems like not working properly
1) I have encountered some situation that goes to `(that.context.state === 'running')` directly. if case seems necessary.
2) Since it is "falling back", I was expecting to see feature preparation for test file linked to <audio> when I trigger `that.initializeMicrophoneSampling();`. However, the audio file input had some issue. I believe it is misconfigured.